src/Native/LdaNative/lda_engine.cpp

```diff
@@ -123,20 +123,11 @@ namespace lda {
     LdaEngine::~LdaEngine()
     {
         //delete memory space
-        delete process_barrier_;
-        process_barrier_ = nullptr;
-
-        delete data_block_;
-        data_block_ = nullptr;
-
-        delete atomic_stats_;
-        atomic_stats_ = nullptr;
-
-        delete model_block_;
-        model_block_ = nullptr;
-
-        delete samplerQueue_;
-        samplerQueue_ = nullptr;
+        process_barrier_.reset(nullptr);
+        data_block_.reset(nullptr);
+        atomic_stats_.reset(nullptr);
+        model_block_.reset(nullptr);
+        samplerQueue_.reset(nullptr);
```

Comment:
📝 These calls to `reset` are _likely_ not required, but by including them it ensures the objects are destroyed in the same order the old code destroyed them.